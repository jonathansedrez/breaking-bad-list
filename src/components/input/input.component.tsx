import React, { InputHTMLAttributes } from 'react';
import './input.styles.scss';

interface IInput extends InputHTMLAttributes<HTMLInputElement> {
  value?: string;
}

export const Input: React.FC<IInput> = (props: IInput) => {
  const { value } = props;
  return <input className="input__text" {...props} value={value} />;
};
