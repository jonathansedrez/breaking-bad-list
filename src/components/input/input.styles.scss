@import '../../layout/variables.scss';

.input__text {
  width: 550px;
  padding: 16px 8px;
  font-size: 1.2rem;
  border-radius: 4px;
  border: 1px solid $presentation;
  color: $black;
  font-family: 'Montserrat', sans-serif;

  &::placeholder {
    color: $grey;
    font-weight: 600;
  }
}
