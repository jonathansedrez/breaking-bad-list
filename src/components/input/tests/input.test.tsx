import { render } from '@testing-library/react';

import { Input } from '../input.component';

describe('List Characters screen', () => {
  it('should execute snapshot', () => {
    const { getByRole } = render(<Input />);
    const element = getByRole('textbox');
    // debug();
    expect(element).toMatchSnapshot();
  });
});
