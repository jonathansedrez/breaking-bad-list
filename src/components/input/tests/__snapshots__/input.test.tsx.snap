// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Characters screen should execute snapshot 1`] = `
<input
  class="input__text"
  value=""
/>
`;
